base/collection.go

```diff
@@ -1041,7 +1041,6 @@ func (c *Collection) GetCollectionID() (uint32, error) {
 	}
 	wg.Wait()
```

Comment:
Given there's only 1 path through this code (via the callback) that's async, do we need to `Add`, `Done` and `Wait` all on the same goroutine? Doesn't make much sense to me.

I.e. Can we rewrite this bit of code so that we only add to the WaitGroup when we pass the `err != nil` check - and then do the `Add`/`Wait` only for the callback's `Done`?

/cc @torcolvin

Reply:
If we add the waitgroup before `err != nil` check, there's a chance we might end up with negative wait group I think.

The other pattern we could use is the signal pattern, but I'm not sure which is more readable. https://github.com/couchbase/sync_gateway/blob/335e13d55d6b6ec64d74ecee440a66842ce526f5/base/collection.go#L1074

Reply:
as discussed - let's just move forward with the current fix